### ISSUE TITLE
linkedin_oauth2: to_str not backwards compatible with v1 data

### ISSUE THREAD

**Issue by pennersr** [pennersr/django-allauth#2220]
When running 0.39.0 against a database containing pre 0.39.0 `linkedin_oauth2` accounts:

```
AttributeError: 'str' object has no attribute 'get'
(6 additional frame(s) were not displayed)
...
  File "allauth/account/views.py", line 97, in get
    self.request, response, form=form, data=self._get_ajax_data_if())
  File "allauth/account/views.py", line 120, in _get_ajax_data_if
    if get_adapter(self.request).is_ajax(self.request)
  File "allauth/socialaccount/views.py", line 121, in get_ajax_data
    'name': provider_account.to_str()
  File "allauth/socialaccount/providers/linkedin_oauth2/provider.py", line 41, in to_str
    first_name = _extract_name_field(self.account.extra_data, 'firstName')
  File "allauth/socialaccount/providers/linkedin_oauth2/provider.py", line 13, in _extract_name_field
    localized = v.get('localized', {})

AttributeError: 'str' object has no attribute 'get'
```

**Comment by pennersr** [comment]
Fixed in 318fdef8e299dd6280cbdfc0680fd38549e3811f

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/2220
